erdjs/src/dapp/hwProvider.ts

```diff
@@ -90,15 +94,26 @@ export class HWProvider implements IDappProvider {
 
         const address = await this.getCurrentAddress();
         transaction.sender = new Address(address);
-
-        const sig = await this.hwApp.signTransaction(transaction.serializeForSigning(new Address(address)));
+        let signWithHash = await this.shouldSignWithHash();
+        const sig = await this.hwApp.signTransaction(transaction.serializeForSigning(new Address(address)), signWithHash);
         transaction.signature = new Signature(sig);
 
         await transaction.send(this.provider);
 
         return transaction;
     }
 
+    private async shouldSignWithHash(): Promise<boolean> {
+        if (!this.hwApp) {
+            throw new Error("HWApp not initialised, call init() first");
```

Comment:
You can also define specific errors in `errors.ts` - not necessarily in this case, but in general. Fine as it is.